### ISSUE TITLE
Unexpected file auto-reformatting when 'Automatically decode entities' is checked

### ISSUE THREAD

**Issue by xomx** [rdipardo/nppHTMLTag#12]
I was bitten by the following, when updating my translation file.
IDK if it is a feature or bug and can be prevented somehow(?).

STR (using fresh x64 N++ portable v8.7.4):
- HTMLTag installed via PluginAdmin (v1.5.1)
- in N++ open `.\localization\english.xml`
- go to item id 6910 there ( Ctrl+F for `<Item id="6910" name="Minimum Size for Auto-Checking &quot;In selection":"/>` )
- move cursor at the EOL:
  ![HTMLTag-plugin-before](https://github.com/user-attachments/assets/bad1e77d-df4b-4c14-9b0c-a85e41dfa3ee)
- press Enter:
  ![HTMLTag-plugin-after](https://github.com/user-attachments/assets/4bae3c7e-c025-41f0-a705-ce14b7c40fa1)

**Comment by rdipardo** [comment]
<del>I can't tell from the image but I'm supposing the EOL mode is <kbd>CRLF</kbd>, right?</del>

<del>The [`SCN_CHARADDED` hook][0] has puzzled me for a long time; `TAB` does not appear to trigger it, yet it fires *twice* for DOS EOLs: once for `\r`, and again for `\n` 🤔 </del>

Never mind; the real issue was a too-greedy backtracking loop: 6e3800c

[0]: https://github.com/rdipardo/nppHTMLTag/blob/12bac1ab231839962436d4b7ef5be0b241992f3f/src/HtmlTag.cpp#L186

**Comment by rdipardo** [comment]
Should be fixed now: <https://app.circleci.com/pipelines/github/rdipardo/nppHTMLTag/18/workflows/05a1cc7f-a1a8-4f9d-b3a3-cf7ca31db307/jobs/19/artifacts>

*Note: build artifacts expire in 30 days*

**Comment by xomx** [comment]
Tested - ok.
Thanks for the quick fix and this useful plugin!

**Comment by xomx** [comment]
Eee, no, I spoke too soon. In the 

> Should be fixed now: https://app.circleci.com/pipelines/github/rdipardo/nppHTMLTag/18/workflows/05a1cc7f-a1a8-4f9d-b3a3-cf7ca31db307/jobs/19/artifacts

is no new `HTMLTag.dll` binary file, only ini!
So I actually tested that english.xml file without any HTMLTag plugin behind...

Edit: Only arm64 zip contains the binary.

**Comment by rdipardo** [comment]
@xomx 

> Only arm64 zip contains the binary.

Yep. The x86 Intel builds were targeting XP. It seems at some time in the past two weeks the XP v141 toolset was pulled by Microsoft (with no announcement :thinking:?) and then Circle CI upgraded their image to pull in the newest VS Studio release.

No plans on reverting to Free Pascal for the sake of XP, though. Plugin versions 1.4.4 and older will even run on NT 4.0, yes, but Pascal's memory model is [broke-ass][1] and the code looks like a dog's breakfast.

No harm done since my earlier "fix" actually had a [major regression][2] in it.

Newest builds are here: <https://app.circleci.com/pipelines/github/rdipardo/nppHTMLTag/19/workflows/42899453-91c3-4c69-8496-c2287fb12936/jobs/20/artifacts>

[1]: https://gitlab.com/freepascal.org/fpc/source/-/issues/41023
[2]: https://github.com/rdipardo/nppHTMLTag/commit/55d5d224f3890c77d5f65667889fce33a2925d20

**Comment by xomx** [comment]
@rdipardo 
Works ok, thanks again.

**Comment by rdipardo** [comment]
Issues without a patched version on the Plugin List can stay open in case other users come here to report the same thing.

**Comment by rdipardo** [comment]
Fixed in *HTML Tag* [1.5.2], available *via* [Plugins Admin], if you are using Notepad++ [8.7.6](https://notepad-plus-plus.org/downloads/v8.7.6) or newer.

[Plugins Admin]: https://npp-user-manual.org/docs/plugins/#install-using-plugins-admin
[1.5.2]: https://bitbucket.org/rdipardo/htmltag/src/v1.5.2/NEWS.textile
